doc/source/security/data-at-rest-encryption.rst

```diff
@@ -166,6 +171,52 @@ after creating the encrypted tablespace. If you are using Master key encryption,
+
+If you set the ``secret_mount_point_version`` to ``1`` but the path pointed by ``secret_mount_point`` is based on ``KV Secrets Engine - Version 2 (kv-v2)``, the plugin initialization succeeds but any MySQL keyring-related operations fail.
+
+.. list-table::
+  :widths: 10 40
+  :header-rows: 1
+
+  * - Value
+    - Description
+  * - 1
+    - Works with ``KV Secrets Engine - Version 1 (kv)``. When forming key operation URLs, the ``secret_mount_point`` is always used without any transformations.
+
+      For example, to return a key named ``skey``, the URL is <vault_url>/v1/<secret_mount_point>/skey
+  * - 2
+    - Works with ``KV Secrets Engine - Version 2 (kv)` The initialization logic splits the ``secret_mount_point`` parameter into two parts: ``mount_point_path`` (the mount path under which the Vault Server secret was created), and ``directory_path`` (a virtual directory suffix that can be used to create virtual namespaces with the same real mount point).
+
+      Both the ``mount_point_path`` and the ``directory_path`` are needed to form key access URLs. For example, <vault_url>/<v1>/<mount_point_path/data/<directory_path>/skey
+  * - Auto
+    - An auto-detection mechanism probes and determines if the secrets engine version is ``kv`` or ``kv-v2`` and based on the outcome will either use the ``secret_mount_point`` as is, or split the ``secret_mount_point`` into two parts.
+  * - None 
+    - If the ``secret_mount_point_version`` is not set, the behavior is the same as if you had set the value to ``Auto``. 
+
+Upgrading from 5.7.32 or earlier to 5.7.33 or later
+-----------------------------------------------------
+
+The ``keyring_vault`` plugin configuration files created before |Percona Server| 5.7.33 work only with ``KV Secrets Engine - Version 1 (kv)`` and do not have the ``secret_mount_point_version`` parameter. After the upgrade to 5.7.33 or later, the ``secret_mount_point_version`` is implicitly considered ``Auto`` and the information is probed and the secrets engine version is determined to ``1``.
+
+Upgrading from Vault Secrets Engine Version 1 to Version 2 
+-----------------------------------------------------------
+
+You can upgrade from the Vault Secrets Engine Version 1 to Version 2, use either of the following methods:
+
+* Set the ``secret_mount_point_version`` to ``Auto`` or not set in the ``keyring_vault`` plugin configuration files in all Percona Servers. This setting ensures the auto-detection mechanism is invoked during the plugin initialization. 
+
+* Set the ``secret_mount_point_version`` to ``2`` to ensure that plugins do not initialize unless the ``kv`` to ``kv-v2`` upgrade completes.
+
+.. note:: the ``keyring_vault`` plugin that works with ``kv-v2`` secrect engines do not use built-in key versioning capabilities. The keyring key versions are encoded into key names.
+
+
+
 System Variables
```

Comment:
@hrvojem also asked to add a section about considerations when doing major (5.7 to 8.0) Percona Server upgrades.
* When upgrading from <=5.7.32, you are obviously using `KV Secrets Engine Version 1` (`kv`) and you can upgrade to any version of 8.0. Both old and new `keyring_vault` plugin versions will be working correctly with existing Vault Server data under existing `keyring_vault` plugin configuration file.
* When upgrading from >=5.7.33, there are 2 options:
  * If you are using `KV Secrets Engine Version 1` (`kv`), you can upgrade to any version of 8.0.
  * If you are using `KV Secrets Engine Version 2` (`kv-v2`), you need to upgrade to Percona Server >=8.0.23 (not yet released), because 8.0.23 is the first version on the 8.0 series that has `keyring_vault` plugin that supports `kv-v2`.

Reply:
Added.